tests_new/integration_tests/modules/share_base/test_new_crossacc_s3_share.py

```diff
@@ -0,0 +1,393 @@
+import pytest
+from assertpy import assert_that
+
+from tests_new.integration_tests.aws_clients.athena import AthenaClient
+from dataall.modules.shares_base.services.shares_enums import (
+    ShareItemStatus,
+    ShareObjectStatus,
+    ShareItemHealthStatus,
+    ShareableType,
+)
+from tests_new.integration_tests.modules.s3_datasets.aws_clients import S3Client
+from tests_new.integration_tests.modules.share_base.conftest import clean_up_share
+from tests_new.integration_tests.modules.share_base.queries import (
+    create_share_object,
+    submit_share_object,
+    add_share_item,
+    get_share_object,
+    reject_share_object,
+    approve_share_object,
+    revoke_share_items,
+    delete_share_object,
+    verify_share_items,
+    update_share_request_reason,
+    update_share_reject_reason,
+    get_s3_consumption_data,
+)
+from tests_new.integration_tests.modules.share_base.utils import (
+    check_share_ready,
+    check_share_items_verified,
+    get_group_session,
+    get_role_session,
+)
+
+
+@pytest.mark.parametrize(
+    'principal_type',
+    ['Group', 'ConsumptionRole'],
+)
+def test_create_and_delete_share_object(
+    client5, persistent_cross_acc_env_1, session_s3_dataset1, consumption_role_1, group5, principal_type
+):
```

Comment:
Instead of parametrizing the tests you can [parametrize a fixture](https://docs.pytest.org/en/latest/how-to/fixtures.html#parametrizing-fixtures) and then all tests that are using this fixture will run as many times as the parameters of the fixture.

For example in this case you can do something like...

```
@pytest.fixture(params=["Group", "ConsumptionRole"])
def principal1(request, group5, consumption_role_1):
    """
    :return: tuple with (principalUri, principalType)
    """
    if request.param is 'Group':
        yield (group5, request.param)
    else:
        yield (consumption_role_1.consumptionRoleUri, request.param)
```

Reply:
implemented